src/CTA.Rules.Actions/ActionsLoader.cs

```diff
@@ -47,138 +47,137 @@ public ActionsLoader(List<string> assemblyPaths)
 
             var actionsAssembly = AppDomain.CurrentDomain.GetAssemblies().Where(a => a.FullName.Contains("CTA.Rules.Actions")).FirstOrDefault();
 
-            try
-            {
-                LoadAssemblyTypes(actionsAssembly);
-            }
-            catch (Exception ex)
+            var assemblies = new List<Assembly>
             {
-                LogHelper.LogError(ex, "Error while loading actions assembly");
-                throw ex;
-            }
+                actionsAssembly
+            };
+
             foreach (var path in assemblyPaths)
             {
                 try
                 {
                     var assembly = AssemblyLoadContext.Default.LoadFromAssemblyPath(path);
-                    LoadAssemblyTypes(assembly);
+                    assemblies.Add(assembly);
                 }
                 catch (Exception ex)
                 {
-                    LogHelper.LogError(ex, string.Format("Error loading actions from {0}{1}{2}", path, Environment.NewLine, ex.Message));
+                    LogHelper.LogError(ex, string.Format("Error loading assembly from {0}{1}{2}", path, Environment.NewLine, ex.Message));
                 }
             }
-        }
```

Comment:
Was this a refactor outside of the new config file suggestions? If so, in the future it may be helpful to separate manual refactoring changes from automated ones, esp since this new format modified several files. Refactor here looks good regardless.

Reply:
Had to make the private fields readonly, and to do this properly, i had to refactor this part.

Reply:
Ah, I see. That makes sense- thanks for the context